src/main/java/it/gov/pagopa/pu/organization/service/taxonomy/TaxonomySynchronizationService.java

```diff
@@ -1,30 +1,81 @@
 package it.gov.pagopa.pu.organization.service.taxonomy;
 
 import it.gov.pagopa.pu.organization.connector.pagopapayments.TaxonomyServiceImpl;
+import it.gov.pagopa.pu.organization.mapper.TaxonomyMapper;
+import it.gov.pagopa.pu.organization.model.Taxonomy;
 import it.gov.pagopa.pu.organization.repository.TaxonomyRepository;
-import it.gov.pagopa.pu.pagopapayments.dto.generated.Taxonomy;
+import it.gov.pagopa.pu.pagopapayments.dto.generated.TaxonomyDTO;
+import jakarta.transaction.Transactional;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 @Service
 @Slf4j
 public class TaxonomySynchronizationService {
   private final TaxonomyRepository taxonomyRepository;
   private final TaxonomyServiceImpl pagopaPaymentsClient;
 
+  private final TaxonomyMapper taxonomyMapper;
+
   public TaxonomySynchronizationService(
     TaxonomyRepository taxonomyRepository,
-    TaxonomyServiceImpl pagopaPaymentsClient
+    TaxonomyServiceImpl pagopaPaymentsClient, TaxonomyMapper taxonomyMapper
   ) {
     this.taxonomyRepository = taxonomyRepository;
     this.pagopaPaymentsClient = pagopaPaymentsClient;
+    this.taxonomyMapper = taxonomyMapper;
   }
+
+  @Transactional
   public void synchronizeTaxonomies(String accessToken) {
     log.info("TaxonomySynchronizationService :: Synchronizing taxonomy");
-    List<Taxonomy> taxonomies = pagopaPaymentsClient.fetchTaxonomy(accessToken);
-    //todo business logic to be implemented in P4ADEV-2150
+    List<TaxonomyDTO> fetchedTaxonomies = pagopaPaymentsClient.fetchTaxonomy(accessToken);
+
+    List<Taxonomy> existingTaxonomies = taxonomyRepository.findAll();
+    Map<String, Taxonomy> existingTaxonomyMap = new HashMap<>();
+    if(existingTaxonomies != null) {
+      existingTaxonomyMap = existingTaxonomies.stream()
+        .collect(Collectors.toMap(Taxonomy::getTaxonomyCode, taxonomy -> taxonomy));
```

Comment:
does it compile if you use the following?
```suggestion
        .collect(Collectors.toMap(Taxonomy::getTaxonomyCode, Function::identity));
```